### DISCUSSION TITLE
API Usage limit

### DISCUSSION THREAD

**Discussion by w1lldabea5t** [GrandyB/SheetsIO#53] [Q&A]
Does the approx 300 requests per minute refer to individual cells, or the sheet in general?
Say i reference 50 cells to update, and the read request fires off every 2 seconds, is this going to hit the 300 cap in 12 seconds?
Or is the amount of cells read ignored and it's just the amount of times the entire sheet is read?

**Comment by GrandyB** [comment]
It's a request rate limit.

A request can ask for as many cells within a single tab in a sheet as it likes, it makes no difference... but it's a limit of 100 overall requests in 100 seconds per API key I believe.

Worth looking up the latest limits.

**Reply by w1lldabea5t** [reply]
![image](https://github.com/GrandyB/SheetsIO/assets/145459150/8c0a0db4-b201-483b-9bb4-f7ee596d82ab)

ah yes, the current limit is 100 requests within 100 seconds
I've currently stuck to the set default of a 2 second refresh, but only using a few cells to reference.
I'm looking at adding a few hundred cells so I can bring in these score sheets and was worried it'd instantly hit the limit
